plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -577,9 +602,16 @@ public void alterTransactionalTable(Table table, long transactionId, long writeI
         table.setWriteId(writeId);
         checkArgument(writeId >= table.getWriteId(), "The writeId supplied %s should be greater than or equal to the table writeId %s", writeId, table.getWriteId());
         AlterTableRequest request = new AlterTableRequest(table.getDbName(), table.getTableName(), table);
-        request.setValidWriteIdList(getValidWriteIds(ImmutableList.of(format("%s.%s", table.getDbName(), table.getTableName())), transactionId));
+        request.setValidWriteIdList(getTableWriteIds(table.getDbName(), table.getTableName(), transactionId));
         request.setWriteId(writeId);
         request.setEnvironmentContext(environmentContext);
+        log.info("In alterTransactionalTable, tableWriteId %s, validWriteIds %s", table.getWriteId(), request.getValidWriteIdList());
         client.alter_table_req(request);
     }
+
+    private String getTableWriteIds(String databaseName, String tableName, long transactionId)
```

Comment:
Seems we should collect this at the beginning of the operation, not when persisting the results. See also comment above

Reply:
Yes, fixed per the comment above.

Reply:
(per above, i don't see where)

Reply:
Hopefully this is settled by now.